File: frontend/src/app/shared/service/admin/prof/TrancheHoraireProfAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {TrancheHoraireProfDto} from 'src/app/shared/model/prof/TrancheHoraireProf.model';
import {TrancheHoraireProfCriteria} from 'src/app/shared/criteria/prof/TrancheHoraireProfCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class TrancheHoraireProfAdminService extends AbstractService<TrancheHoraireProfDto, TrancheHoraireProfCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): TrancheHoraireProfDto {
        return new TrancheHoraireProfDto();
    }

    public constrcutCriteria(): TrancheHoraireProfCriteria {
        return new TrancheHoraireProfCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/trancheHoraireProf/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/prof/TrancheHoraireProfAdminService.java
package ma.zs.alc.service.facade.admin.prof;

import java.util.List;
import ma.zs.alc.bean.core.prof.TrancheHoraireProf;
import ma.zs.alc.dao.criteria.core.prof.TrancheHoraireProfCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface TrancheHoraireProfAdminService extends  IService<TrancheHoraireProf,TrancheHoraireProfCriteria>  {

    List<TrancheHoraireProf> findByProfId(Long id);
    int deleteByProfId(Long id);
    long countByProfRef(String ref);




}
